contracts/test/Staking.t.sol

```diff
@@ -0,0 +1,1400 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.30;
+
+import {Test} from "@forge-std/Test.sol";
+import {Staking, Ownable} from "../src/Staking.sol";
+import {ERC20, IERC20Errors} from "@oz/token/ERC20/ERC20.sol";
+
+contract MockERC20 is ERC20 {
+    constructor(string memory name, string memory symbol) ERC20(name, symbol) {}
+
+    function mint(address to, uint256 amount) external {
+        _mint(to, amount);
+    }
+}
+
+contract StakingTest is Test {
+    Staking public staking;
+    MockERC20 public token;
+
+    address public owner = address(0x1);
+    address public validator = address(0x2);
+    address public staker = address(0x3);
+    address public staker2 = address(0x4);
+    address public other = address(0x5);
+
+    uint128 public constant INITIAL_WITHDRAW_DELAY = 1 days;
+    uint256 public constant CONFIG_TIME_DELAY = 2 days;
+
+    event StakeIncreased(address indexed staker, address indexed validator, uint256 amount);
+    event WithdrawalInitiated(
+        address indexed staker, address indexed validator, uint64 indexed withdrawalId, uint256 amount
+    );
+    event WithdrawalClaimed(address indexed staker, address indexed validator, uint256 amount);
+    event ValidatorsProposed(
+        bytes32 indexed validatorsHash, address[] validator, bool[] isRegistration, uint256 executableAt
+    );
+    event ValidatorUpdated(address indexed validator, bool isRegistered);
+    event WithdrawDelayProposed(uint256 currentDelay, uint256 proposedDelay, uint256 executableAt);
+    event WithdrawDelayChanged(uint256 oldDelay, uint256 newDelay);
+    event TokensRecovered(address indexed token, address indexed to, uint256 amount);
+
+    function setUp() public {
+        vm.startPrank(owner);
+        token = new MockERC20("Safe Token", "SAFE");
+        staking = new Staking(owner, address(token), INITIAL_WITHDRAW_DELAY, CONFIG_TIME_DELAY);
+
+        // Register validator
+        address[] memory validators = new address[](1);
+        validators[0] = validator;
+        bool[] memory isRegistration = new bool[](1);
+        isRegistration[0] = true;
+
+        staking.proposeValidators(validators, isRegistration);
```

Comment:
I really wish this was `(address, bool)[]` instead of `(address[], bool[])`.

Not related to this PR - just something that I noticed that bothered me.